02 get topics

zairacodes opened the pull request:
Hello!

Thanks for taking a look at my code.

I've added an endpoint for GET api/topics
In particular, I have:
- created and .gitignored a playground.sql file to help me visualise my test database
- installed the necessary packages and set up my test, app and listen files
- written and failed tests for GET api/topics, both happy and sad paths
- implemented app.js and the relative controller and model
- successfully passed the tests

nc-hannah-burks reviewed (commented):
A great start here Zaira!

Some things to make note of: 

- For your 400 error, you may look to use a generic 404 error, so we can be sure we only fall into that block of code if there is an invalid path
https://notes.northcoders.com/courses/js-back-end/error-handling#5-generic-404-errors

- It would be ideal to add some status 500 (server error) handling here.

zairacodes commented:
Thank you Hannah!

Following your feedback, I have just:
- fixed my test for a 404 error (I wrote 400 in the test description by mistake, but was correctly testing for 404)
- added an error handling middleware which will handle server errors as well
